tests/compilation/test_repetitions.py

```diff
@@ -0,0 +1,257 @@
+# Copyright 2024 PsiQuantum, Corp.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+
+import pytest
+from qref import SchemaV1
+from qref.schema_v1 import RoutineV1
+
+from bartiq import compile_routine, evaluate
+from bartiq.errors import BartiqCompilationError
+
+
+def _routine_with_repetition(repetition_dict: dict) -> RoutineV1:
+    return SchemaV1(
+        program={
+            "name": "root",
+            "input_params": ["unit_prob", "unit_T"],
+            "children": [
+                {
+                    "name": "child",
+                    "resources": [
+                        {"name": "T", "type": "additive", "value": "unit_T"},
+                        {"name": "success_rate", "type": "multiplicative", "value": "unit_prob"},
+                    ],
+                }
+            ],
+            "repetition": repetition_dict,
+        },
+        version="v1",
+    )
+
+
+def _constant_sequence_sum(unit_cost, count, multiplier):
+    sum = 0
+    for _ in range(count):
+        sum += unit_cost * multiplier
+    return sum
```

Comment:
**Question**: Why do we write this explicitly? Isn't it as simple as `return count * unit_cost * multiplier`?

The same question for other sequences in this file.

Reply:
I just wanted this to be dead simple and explicit.